### PR TITLE
Fix `add`, introduce `isEmpty`

### PR DESCRIPTION
**Pull request by ladd**
- Embarassingly, the metric 'add' operation was broken, and would add twice the provided number on first call. Luckily, we weren't using it yet.
- Add a new `isEmpty` check to simplify filtering zero metrics in the client's reporter
- Add a bunch of (mostly Claude-written) tests
- File organization
- Other minor tweaks and fixes

@jparise 
@bachand

### CODE REVIEW

File: Sources/NautilusTelemetry/Metrics/HistogramValues.swift [hunk 1]
```diff
@@ -26,6 +26,11 @@ struct HistogramBuckets<T: MetricNumeric> {
 	var data: [UInt64]
 	let explicitBounds: [T]
 
+	var isEmpty: Bool {
+		guard !data.isEmpty else { return true }
+		return data.allSatisfy { $0 == 0 }
```
Comment (jparise) [inline Sources/NautilusTelemetry/Metrics/HistogramValues.swift:31]: This might be more clear:

```suggestion
		return data.isEmpty || data.allSatisfy { $0 == 0 }
```

**Review by jparise — APPROVED** [review]
Tests are pretty verbose. AI generated?

**Comment by ladd** [comment]
> Tests are pretty verbose. AI generated?

Yep. It went a little overboard, but had some OK boundary condition tests.

File: Sources/NautilusTelemetry/Metrics/HistogramValues.swift (hunk 1 above)
Comment (ladd) [reply Sources/NautilusTelemetry/Metrics/HistogramValues.swift:31]: Sure.

File: Sources/NautilusTelemetry/Metrics/ObservableGauge.swift [hunk 2]
```diff
@@ -28,6 +28,8 @@ public class ObservableGauge<T: MetricNumeric>: Instrument, ExportableInstrument
 	public private(set) var endTime: ContinuousClock.Instant? = nil
 	public let aggregationTemporality = AggregationTemporality.unspecified
 
+	public var isEmpty: Bool { false }
```
[the remark this answers is not in the document]
Comment (bachand) [reply Sources/NautilusTelemetry/Metrics/ObservableGauge.swift:31]: That makes sense @ladd. To confirm my understanding, would updating this code to be the following also be valid, since we're saying that it's empty if there are no values but still sending 0 values?

```swift
	public var isEmpty: Bool { values.isEmpty }
```
